src/output.h

```diff
@@ -25,39 +26,43 @@
 #ifndef _OUTPUT_H
 #define _OUTPUT_H
 
+#include <set>
+#include <string>
+
 #include <glib.h>
 #include "song-meta-data.h"
 
-// Feedback for the controlling part what is happening with the
-// output.
-enum PlayFeedback {
-  PLAY_STOPPED,
-  PLAY_STARTED_NEXT_STREAM,
-};
-typedef void (*output_transition_cb_t)(enum PlayFeedback);
-
-// In case the stream gets to know details about the song, this is a
-// callback with changes we send back to the controlling layer.
-typedef void (*output_update_meta_cb_t)(const struct SongMetaData *);
-
-int output_init(const char *shortname);
-int output_add_options(GOptionContext *ctx);
-void output_dump_modules(void);
-
-int output_loop(void);
-
-void output_set_uri(const char *uri, output_update_meta_cb_t meta_info);
-void output_set_next_uri(const char *uri);
-
-int output_play(output_transition_cb_t done_callback);
-int output_stop(void);
-int output_pause(void);
-int output_get_position(gint64 *track_dur_nanos, gint64 *track_pos_nanos);
-int output_seek(gint64 position_nanos);
-
-int output_get_volume(float *v);
-int output_set_volume(float v);
-int output_get_mute(int *m);
-int output_set_mute(int m);
+namespace Output {
+typedef enum OutputState { kPlaybackStopped, kStartedNextStream } OutputState;
```

Comment:
No need for typedef. The enum type can be used as-is.

If we make it an enum class, we also get the added benefit that it requires scoped use.
```
enum class OutputState {.... };
```

Given that we are already in the namespace `Output`, maybe just call it `State` ? Then references to it read pretty natural as `Output::State::kPlaybackStopped`.

Reply:
Yeah you're right. Output as a prefix is superfluous.